synfig-studio/src/synfigapp/actions/waypointsetsmart.cpp

```diff
@@ -162,13 +177,10 @@ Action::WaypointSetSmart::set_param(const synfig::String& name, const Action::Pa
 bool
 Action::WaypointSetSmart::is_ready()const
 {
-	if(!value_node)
-		synfig::error("Missing value_node");
-
```

Comment:
You should check if either `value_node` or `value_desc` is set.

Reply:
hmm well none are explicitly needed for the action now. It used to be that a value_node was needed for it, but now in the case that there is no value_node it is made when we redirect to the value desc set action.

Reply:
It is. Without `value_desc` set, it is impossible to call ValueDescSet action.
So either valuenode or valudesc must be set.

Reply:
Fair point. I meant that not always are we going to be calling the ValueDescSet action, as far as I remember it only needs to be called if we are creating the initial waypoint for the parameter. But fair enough I see your point.

Reply:
That's why we need to check if one of them (wither `value_node` or `value_desc`) is set.
We cannot proceed if we don't have any of these info.

Reply:
oh yes definitely, we need at least one of the to proceed.